python/lsst/ip/isr/crosstalk.py

```diff
@@ -717,20 +720,31 @@ def run(self, exposure, crosstalk=None,
 
             if crosstalk.interChip:
                 if crosstalkSources:
+                    # Parse crosstalkSources: Identify which detectors we have available
+                    if isinstance(crosstalkSources[0], lsst.afw.image.Exposure):
+                        # Received afwImage.Exposure
+                        sourceNames = [exp.getDetector().getName() for exp in crosstalkSources]
+                    else:
+                        # Received dafButler.DeferredDatasetHandle
+                        detectorList = [source.dataId['detector'] for source in crosstalkSources]
+                        sourceNames = [camera[detector].getName() for detector in detectorList]
+
                     for detName in crosstalk.interChip:
-                        if isinstance(crosstalkSources[0], 'lsst.afw.image.Exposure'):
-                            # Received afwImage.Exposure
-                            sourceNames = [exp.getDetector().getName() for exp in crosstalkSources]
-                        else:
-                            # Received dafButler.DeferredDatasetHandle
-                            sourceNames = [expRef.get(datasetType='isrOscanCorr').getDetector().getName()
-                                           for expRef in crosstalkSources]
                         if detName not in sourceNames:
                             self.log.warn("Crosstalk lists %s, not found in sources: %s",
                                           detName, sourceNames)
                             continue
+                        # Get the coefficients.
                         interChipCoeffs = crosstalk.interChip[detName]
+
                         sourceExposure = crosstalkSources[sourceNames.index(detName)]
+                        if not isinstance(sourceExposure, lsst.afw.image.Exposure):
+                            # Dereference the dafButler.DeferredDatasetHandle.
+                            sourceExposure = sourceExposure.get()
+
+                        self.log.warn("Correcting detector %s with ctSource %s",
```

Comment:
Is `warn` the best level here? This seems like a good thing to me 😁

Reply:
Updated to `info`, because that should be less confusing.